internal/osbuild/shell_init_stage.go

```diff
@@ -0,0 +1,53 @@
+package osbuild
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const filenameRegex = "^[a-zA-Z0-9\\.\\-_]{1,250}$"
+const envVarRegex = "^[A-Z][A-Z0-9_]*$"
+
+type ShellInitStageOptions struct {
+	Files map[string]ShellInitFile `json:"files"`
+}
+
+func (ShellInitStageOptions) isStageOptions() {}
+
+func (options ShellInitStageOptions) validate() error {
+	fre := regexp.MustCompile(filenameRegex)
+	vre := regexp.MustCompile(envVarRegex)
+	for fname, kvs := range options.Files {
+		if !fre.MatchString(fname) {
+			return fmt.Errorf("filename %q doesn't conform to schema (%s)", fname, filenameRegex)
+		}
+
+		for _, kv := range kvs.Env {
+			if !vre.MatchString(kv.Key) {
+				return fmt.Errorf("variable name %q doesn't conform to schema (%s)", kv.Key, envVarRegex)
+			}
+		}
+	}
```

Comment:
This code does not validate that there is at least one key/value defined as required by 
https://github.com/osbuild/osbuild/blob/d3229dc929d32e927eae4606182c5b02553181ad/stages/org.osbuild.shell.init#L28

Reply:
Good catch.